functional-tests/test-bed/src/test/java/org/arquillian/smart/testing/ftest/affected/HistoricalChangesAffectedTestsSelectionExecutionFunctionalTest.java

```diff
@@ -54,6 +54,64 @@ public void should_only_execute_tests_related_to_multiple_commits_in_business_lo
 
         // then
         assertThat(actualTestResults).containsAll(expectedTestResults).hasSameSizeAs(expectedTestResults);
+
     }
 
+    @Test
+    public void should_execute_selected_tests_when_smart_testing_is_enabled() throws Exception {
+        // given
+        project.configureSmartTesting()
+            .executionOrder(AFFECTED)
+            .inMode(SELECTING)
+            .enable();
+
+        project.applyAsCommits("Single method body modification - sysout",
+                "Inlined variable in a method");
+
+        // when
+        final List<TestResult> actualTestResults = project
+            .buildOptions()
+            .withSystemProperties("git.last.commits", "2", "disableSmartTesting", "false")
+            .configure()
+            .build();
+
+        // then
+        assertThat(actualTestResults).hasSize(3);
+    }
+
+    @Test
+    public void should_execute_all_tests_when_smart_testing_is_disabled() throws Exception {
+        // when
+        final List<TestResult> actualTestResults = project
+            .buildOptions()
+            .withSystemProperties("disableSmartTesting", "true")
+            .configure()
+            .build();
+
+        // then
+        assertThat(actualTestResults).hasSize(77);
+    }
+
+    @Test
+    public void should_execute_all_tests_when_smart_testing_is_disabled_irrespective_of_strategy() throws Exception {
+        // given
+        project.configureSmartTesting()
+            .executionOrder(AFFECTED)
+            .inMode(SELECTING);
+            //.enable();
```

Comment:
Having it done this way will never really bring our extension into the maven build. This test is equal to running `$ mvn clean install -DdisableSmartTesting=true` on any random Maven project. It will run normally as our extension is not even configured. Essentially `.enable()` takes care of adding our extension to the build configuration. Why is it commented out?